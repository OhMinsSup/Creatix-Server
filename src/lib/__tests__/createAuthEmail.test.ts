import faker from 'faker';
import { createAuthEmail } from '../createAuthEmail';

const uuid = faker.random.uuid();

describe('CreateAuthEmail', () => {
  it('Login Email Create HTML', () => {
    expect(createAuthEmail(true, uuid)).toMatchSnapshot();
  });

  it('Register Email Create HTML', () => {
    expect(createAuthEmail(false, uuid)).toMatchSnapshot();
  });
});
