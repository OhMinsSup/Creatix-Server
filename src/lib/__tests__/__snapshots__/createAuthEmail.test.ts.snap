// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`회원 인증 이메일 템플릿 로그인을 했을 때 HTML이 만들어진 경우 1`] = `
Object {
  "body": "<a href=\\"https://velog.io\\"><img src=\\"https://images.velog.io/email-logo.png\\" style=\\"display: block; width: 128px; margin: 0 auto;\\"/></a>
      <div style=\\"max-width: 100%; width: 400px; margin: 0 auto; padding: 1rem; text-align: justify; background: #f8f9fa; border: 1px solid #dee2e6; box-sizing: border-box; border-radius: 4px; color: #868e96; margin-top: 0.5rem; box-sizing: border-box;\\">
        <b style=\\"black\\">안녕하세요! </b>로그인을 계속하시려면 하단의 링크를 클릭하세요. 만약에 실수로 요청하셨거나, 본인이 요청하지 않았다면, 이 메일을 무시하세요.
      </div>
      
      <a href=\\"http://localhost:3000/email-login?code=ebf057f9-cac0-470a-9eb0-ac64f7f17e47\\" style=\\"text-decoration: none; width: 400px; text-align:center; display:block; margin: 0 auto; margin-top: 1rem; background: #845ef7; padding-top: 1rem; color: white; font-size: 1.25rem; padding-bottom: 1rem; font-weight: 600; border-radius: 4px;\\">계속하기</a>
      
      <div style=\\"text-align: center; margin-top: 1rem; color: #868e96; font-size: 0.85rem;\\"><div>위 버튼을 클릭하시거나, 다음 링크를 열으세요: <br/> <a style=\\"color: #b197fc;\\" href=\\"http://localhost:3000/email-login?code=ebf057f9-cac0-470a-9eb0-ac64f7f17e47\\">http://localhost:3000/email-login?code=ebf057f9-cac0-470a-9eb0-ac64f7f17e47</a></div><br/><div>이 링크는 24시간동안 유효합니다. </div></div>",
  "subject": "creatix 로그인",
}
`;

exports[`회원 인증 이메일 템플릿 회원가입을 할 경우에 HTML이 만들어지는 경우 1`] = `
Object {
  "body": "<a href=\\"https://velog.io\\"><img src=\\"https://images.velog.io/email-logo.png\\" style=\\"display: block; width: 128px; margin: 0 auto;\\"/></a>
      <div style=\\"max-width: 100%; width: 400px; margin: 0 auto; padding: 1rem; text-align: justify; background: #f8f9fa; border: 1px solid #dee2e6; box-sizing: border-box; border-radius: 4px; color: #868e96; margin-top: 0.5rem; box-sizing: border-box;\\">
        <b style=\\"black\\">안녕하세요! </b>회원가입을 계속하시려면 하단의 링크를 클릭하세요. 만약에 실수로 요청하셨거나, 본인이 요청하지 않았다면, 이 메일을 무시하세요.
      </div>
      
      <a href=\\"http://localhost:3000/register?code=ebf057f9-cac0-470a-9eb0-ac64f7f17e47\\" style=\\"text-decoration: none; width: 400px; text-align:center; display:block; margin: 0 auto; margin-top: 1rem; background: #845ef7; padding-top: 1rem; color: white; font-size: 1.25rem; padding-bottom: 1rem; font-weight: 600; border-radius: 4px;\\">계속하기</a>
      
      <div style=\\"text-align: center; margin-top: 1rem; color: #868e96; font-size: 0.85rem;\\"><div>위 버튼을 클릭하시거나, 다음 링크를 열으세요: <br/> <a style=\\"color: #b197fc;\\" href=\\"http://localhost:3000/register?code=ebf057f9-cac0-470a-9eb0-ac64f7f17e47\\">http://localhost:3000/register?code=ebf057f9-cac0-470a-9eb0-ac64f7f17e47</a></div><br/><div>이 링크는 24시간동안 유효합니다. </div></div>",
  "subject": "creatix 회원가입",
}
`;
